cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/services/ProgramLifecycleService.java

```diff
@@ -1072,6 +1122,41 @@ private ProgramId validateProgramForRunRecord(String namespaceName, String appNa
   }
 
   /**
+   * Helper method to update log levels for Worker, Flow, or Service.
+   */
+  private void updateLogLevels(ProgramId programId, Map<String, LogEntry.Level> logLevels,
+                               @Nullable String component, @Nullable String runId) throws Exception {
+    ProgramRuntimeService.RuntimeInfo runtimeInfo = findRuntimeInfo(programId, runId).iterator().next();
```

Comment:
So if the program is not running, then it will throw a `NoSuchElementException`, which result in a 500 response? Seems not very good. Shouldn't you check if the return iterator `hasNext` first?